fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -0,0 +1,421 @@
+        index_found = False
+        if self.config.similarity_brain_key is not None:
+            # Load similarity brain run if it exists
+            if (
+                self.config.similarity_brain_key
+                in self.samples._dataset.list_brain_runs()
+            ):
+                index_found = True
+                self._similarity_index = (
+                    self.samples._dataset.load_brain_results(
+                        self.config.similarity_brain_key, load_view=True
+                    )
+                )
+
+                if not len(self._similarity_index._samples) == len(samples):
+                    warnings.warn(
+                        "Passed similarity index is not of the same size as the samples passed. "
+                        "This can cause errors. Make sure the similarity index passed has all "
+                        "of the samples needed "
+                    )
+            else:
+                self._save_similarity_index = True
+
+        # create new similarity index
+        if not index_found:
+            sim_conf_dict_kwargs = {}
+            # if arguments aren't None they take precedence
+            if self.config.similarity_backend is not None:
+                sim_conf_dict_kwargs["name"] = self.config.similarity_backend
+            if self.config.embeddings_field is not None:
+                sim_conf_dict_kwargs[
+                    "embeddings_field"
+                ] = self.config.embeddings_field
+            if self.config.model is not None:
+                sim_conf_dict_kwargs["model"] = self.config.model
+            if self.config.model_kwargs is not None:
+                sim_conf_dict_kwargs["model_kwargs"] = self.config.model_kwargs
+
+            # empty conf if conf dict isn't provided
+            similarity_conf_provided = {}
+            if self.config.similarity_config_dict is not None:
+                similarity_conf_provided = self.config.similarity_config_dict
+            sim_conf_dict_kwargs = {
+                "name": None,  # default if user doesn't provide any value
+                "model": _DEFAULT_MODEL,  # default if user doesn't provide any value
+                **similarity_conf_provided,  # conf over defaults
+                **sim_conf_dict_kwargs,  # arguments over conf
+            }
+
+            backend_name = sim_conf_dict_kwargs.pop("name")
+
+            similarity_config_object = sim._parse_config(
+                backend_name, **sim_conf_dict_kwargs
+            )
+            self._similarity_method = similarity_config_object.build()
+            self._similarity_method.ensure_requirements()
+
+            self._similarity_index = self._similarity_method.initialize(
+                samples, self.config.similarity_brain_key
+            )
+
+    def set_threshold(self, threshold):
+        """Set threshold for leak computation"""
+        self._leak_threshold = threshold
+
+    @property
+    def leaks(self):
```

Comment:
For consistency with `no_leaks_view` here and `duplicates_view` in similarity API, please change this to `leaks_view`